lib/github/pull-request.js

```diff
@@ -1,12 +1,20 @@
 const { Project } = require('../models')
 const transformPullRequest = require('../transforms/pull-request')
 const reduceProjectKeys = require('../jira/util/reduce-project-keys')
+const parseSmartCommit = require('../transforms/smart-commit')
 
 module.exports = async (context, jiraClient, util) => {
   const author = await context.github.users.getForUser({ username: context.payload.pull_request.user.login })
   const { data: jiraPayload } = transformPullRequest(context.payload, author.data)
   const { pull_request: pullRequest } = context.payload
 
+  if (!jiraPayload && context.payload.changes) {
+    const hasIssueKeys = !!parseSmartCommit(context.payload.changes.title.from)
```

Comment:
Is it `title` always an object. Can it be null or undefined? I would check first in order to not crash if it's not there.

We are now parsing the title and the branch name looking for issue keys. We should check the branch name as well.

Reply:
Good thinking! The only downside is the reference would stay if you remove the issue key from the title on a branch with issue keys, but that's probably an acceptable workflow. We already remove the references when a branch is deleted anyway. Will update this.

Reply:
> Is it title always an object. Can it be null or undefined?

I think it cannot be null. We're not supposed to allow blank titles, but I do wonder if there can be other objects off the end of `context.payload.changes`.

Reply:
Added a test case here, but basically the branch issue is already handled since `jiraPayload` will exist if a branch has an issue key.

`context.payload.changes` can have `body` instead of title, so I added check for `changes.title` to handle that case.